### PR TITLE
Add v7 restCompat for invalidating API key with the id field

### PR DESCRIPTION
**Pull request by ywangd**
This PR restore the id field for InvaliateApiKey API so it can be used
if the request explicitly requires v7 compatibility.

Relates: #66671

### CODE REVIEW

**Comment by ywangd** [comment]
The 7.x branch currently does not have any YAML tests to excercise the deprecated `id` field. Because [back then](#66317), YAML test does not allow match warnings with changing content (field position). This is now possible with the new `allowed_warnings_regex` directive. I'll raise a separate PR to add a new YAML test in 7.x so that the deprecated field is tested.

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/rest-compatibility

**Comment by ywangd** [comment]
@elasticmachine update branch

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/apikey/RestInvalidateApiKeyAction.java [hunk 1]
```diff
@@ -78,4 +76,43 @@ public RestResponse buildResponse(InvalidateApiKeyResponse invalidateResp,
     public String getName() {
         return "xpack_security_invalidate_api_key";
     }
+
+    private ConstructingObjectParser<InvalidateApiKeyRequest, Void> getObjectParser(RestRequest request) {
+        if (request.getRestApiVersion() == RestApiVersion.V_7) {
+            final ConstructingObjectParser<InvalidateApiKeyRequest, Void> objectParser = new ConstructingObjectParser<>(
```
Comment (tvernum) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/apikey/RestInvalidateApiKeyAction.java:82]: Is there a reason this isn't a `static final` field like the non-compat parser?
I'm not particularly fussed either way, it just surprised me.

**Review by tvernum — APPROVED** [review]
LGTM

**Pull request merged**